### DISCUSSION TITLE
AWS Permissions

### DISCUSSION THREAD

**Discussion by carlson-matt** [remotely-save/remotely-save#116] [Q&A]
Hello, I have done several searches without finding a good answer through docs and discussions here, but definitely could have missed it so excuse me if this is a duplicate, and feel free to mark as such.

I think it would be beneficial to publish a template aws policy that will allow this service to work, it takes going into the code to find a list of permissions needed, since it is best to limit this to only the permissions needed for it to work. (again, unless this is already documented and I just missed it).

This is the template that I have used for several aws users, i have a user group with these policies attached and use separate users with their own access keys for each obsidian device.

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "ObjectLevelPermissions",
            "Effect": "Allow",
            "Action": [
                "s3:GetObject",
                "s3:DeleteObject",
                "s3:PutObject",
                "s3:ListObjects",
                "s3:GetObjectAttributes"
            ],
            "Resource": "arn:aws:s3:::<bucket>/*"
        },
        {
            "Sid": "BucketLevelPermissions",
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket",
                "s3:HeadBucket",
                "s3:GetBucketLocation"
            ],
            "Resource": "arn:aws:s3:::<bucket>"
        }
    ]
}
```

**Comment by fyears** [comment]
I am sorry but indeed the doc of this plugin doesn’t contain info about aws policies. :-(

One reason is that I am using a **s3-compatible** service, instead of the official Amazon S3. Another reason is that the Amazon policies are kind of complicated.

And I greatly appreciate you share your reference solution here! Thanks!

**Reply by carlson-matt** [reply]
If you're interested I can write a doc that kind of lays out setting the aws permissions to use this app. I think this is a really awesome plugin and would love to contribute in any way needed.

**Comment by fyears** [comment]
the latest doc contains some examples
